Solutions/Marain.ContentManagement.Abstractions/Marain/Cms/CompoundContent.cs

```diff
@@ -0,0 +1,51 @@
+﻿// <copyright file="CompoundContent.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Cms
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// A <see cref="IContentPayload"/> representing content composed from a set of other content, without any particular layout of its own.
+    /// </summary>
+    public class CompoundContent : IContentPayload
```

Comment:
So this is still very much work in progress! Jon sort of hijacked my feature branch and made a PR :)

`ContentSource` is a reference to a specific instance of content at a slug (e.g. version/state/however you are choosing to model it); it basically consists of {Slug,ID}.

`Content` is the common envelope for all content; it has timestamps, author info, title, etc.

`ContentPayload` is the actual "content" in the content, with a known but custom schema.

Reply:
So 'CompoundContent' is indeed an example of the Composite pattern; specifically, it is composition by reference, rather than by aggregation. So `CompositeContent` would not be unreasonable.

It is called `CompoundContent` for largely sentimental reasons (that's what this type was called a long time ago in a galaxy far far away when @HowardvanRooijen and I first addressed this problem.)

Reply:
I find it confusing that we have a type which is

> the actual "content"

and we have a type called `Content`, but that these are different things.

I've created https://github.com/marain-dotnet/Marain.ContentManagement/issues/27 to separate this out from this PR's main intent.